sympy/integrals/integrals.py

```diff
@@ -1243,6 +1242,144 @@ def _sage_(self):
                                     hold=True)
         return f
 
+    def cauchy_principal_value(self, sym, lower_limit=-oo, upper_limit=oo):
```

Comment:
It is not necessary to add sym or limits as they are properties of `self`, in `self.limits`. Instead, there could be `**kwargs` to control the integration.

I also think that the name could be `principal_value`.

Reply:
@jksuom  
>I also think that the name could be principal_value.

I'll change the name to `principal_value`
>Instead, there could be `**kwargs` to control the integration.

How do you propose I implement this? What all do you want a part of `**kwargs`?
(I thought it would be better for the user to just add the limits, and `sym` right there, but if you insist, I'll change it.) What do you say ?

Reply:
It is not necessary to add `sym`, etc. to the parameter list as they are already included as properties of `self`. The `kwargs` can be simply passed on to `integrate(*args, **kwargs)`. They need not be processed here.